server/cherry_pick.go

```diff
@@ -23,6 +23,51 @@ import (
 	"github.com/pkg/errors"
 )
 
+const (
+	cherryPickScheduledMsg = "Cherry pick is scheduled."
+	tooManyCherryPickMsg   = "There are too many cherry picking requests. Please do this manually or try again later."
+)
+
+type cherryPickRequest struct {
+	pr        *model.PullRequest
+	version   string
+	milestone int
+}
+
+func (s *Server) listenCherryPickRequests() {
+	defer func() {
+		close(s.cherryPickStoppedChan)
+	}()
+
+	for job := range s.cherryPickRequests {
+		func() {
+			ctx, cancel := context.WithTimeout(context.Background(), defaultRequestTimeout*time.Second)
+			defer cancel()
+			pr := job.pr
+			cmdOut, err := s.doCherryPick(ctx, strings.TrimSpace(job.version), &job.milestone, pr)
+			if err != nil {
+				msg := fmt.Sprintf("Error trying doing the automated Cherry picking. Please do this manually\n\n```\n%s\n```\n", cmdOut)
+				s.sendGitHubComment(ctx, pr.RepoOwner, pr.RepoName, pr.Number, msg)
+				mlog.Error("Error while cherry picking", mlog.Err(err))
+			}
+		}()
+	}
+}
+
+func (s *Server) finishCherryPickRequests() {
+	close(s.cherryPickRequests)
```

Comment:
`finishCherryPickRequests` is the first thing that's called when `server.Stop` is called.
The first thing we do here is close the channel.

Isn't it possible for a request/event to come in while we're in server shutdown, but still in this function? This would cause the server to still call `handleCherryPick` or `checkIfNeedCherryPick` which writes to this channel which would lead to a panic.

Reply:
Good catch 👍 Reverted it something similar to the my original proposal :)